src/components/Modals/ShareModal/ShareModal.tsx

```diff
@@ -46,19 +48,11 @@ export default class ShareModal extends React.PureComponent<Props, State> {
     }
   }
 
-  handleCopyLink = (e: React.MouseEvent<HTMLAnchorElement>) => {
-    e.preventDefault()
-    if (this.input.current) {
-      const input = this.input.current
-      input.select()
-      document.execCommand('copy')
-      const selection = document.getSelection()
-      if (selection) {
-        selection.removeAllRanges()
-        input.blur()
-        this.setState({ copied: true })
-      }
-      this.props.onShare(ShareTarget.LINK)
+  handleCopyLink = () => {
+    const { copiedTimer } = this.state
+    if (copiedTimer === undefined) {
+      const newCopiedTimer = setTimeout(() => this.setState({ copied: false, copiedTimer: undefined }), 2000)
```

Comment:
We should check the component is still mounted by the time the `setTimeout` is executed, otherwise the `this.setState` will complain. We can do that by setting a flag at the instance level when the component mounts (and unsetting it when it unmounts) and checking against it within the `setTimeout` callback.